src/lib/pubkey/ec_group/ec_scalar.h

```diff
@@ -203,6 +203,11 @@ class BOTAN_PUBLIC_API(3, 6) EC_Scalar final {
       */
       void assign(const EC_Scalar& x);
 
+      /**
+      * Destroy the value stored in this scalar
+      */
+      void zap();
```

Comment:
Should this (and the associated other functions) be named instead `zeroize`? There is a `zap` in secmem.h but I'm not sure this is really standard/common usage.

Also I think the documentation here could be more clear. Does a destructor somewhere get invoked? Can the `EC_Scalar` be reused after calling this function? If it can be reused, is the value after the function returns deterministic?

Maybe

> Zeroize this EC_Scalar.
>
> Equivalent to assigning a zero value, but also does so in a way that attempts to ensure the write always occurs even if a compiler can deduce the assignment is otherwise unnecessary.